src/View/Helper/FormHelper.php

```diff
@@ -1044,6 +1047,19 @@ public function control(string $fieldName, array $options = []): string
         $options = $this->_parseOptions($fieldName, $options);
         $options += ['id' => $this->_domId($fieldName)];
 
+        if (
+            !empty($options['label']) && $options['type'] != 'hidden' ||
+            !empty($options['placeholder']) && $options['type'] != 'hidden'
+        ) {
+            $isFieldError = $this->isFieldError($fieldName);
+            $options += [
+                'aria-label' => !empty($options['label']) ? $options['label'] : $options['placeholder'],
+                'aria-required' => $options['required'] == true ? 'true' : null,
+                'aria-invalid' => $isFieldError ? 'true' : null,
+                'aria-describedby' => $isFieldError ? $options['id'] . '-error' : null,
```

Comment:
Using `$options['id']` won't work, `id` has already been passed through `_domId()`, which transforms the value to include only valid characters, and also might prepend a custom prefix, so it could be different from what is being set in the `error()` method.

You'll need to explicitly pass the field name + suffix through `_domId()`, and do the same in the `error()` method so that both values match.

Reply:
OK, thank You